Feat: Choose Directory and add custom path

0tuedon opened the pull request:
https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/163dab30-e7e9-4a68-aa2e-e00f7660394d

1. You can be able to select a directory from the root
2. You can create a custom path

PENDING
1. No nested directories yet

0tuedon commented:
#105

0tuedon commented:
https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/7cddd9b0-1bd1-47e7-b8c0-fd69dcf2ae22

This was quite challenging mapping out and showing options based on where they are with the directories, @Emmanuel-Develops , i also made sure it does make a request when the nestDir has been saved.
I think this is ready for review

Extheoisah commented on src/components/sideBarContentEdit/SelectField.tsx:
```diff
@@ -15,6 +16,17 @@ type Props = {
+  editedData: string;
+  index: number;
+  path: string;
+  // eslint-disable-next-line no-unused-vars
```
Why do we have this eslint? Is this prop not being used?

0tuedon replied:
> Why do we have this eslint? Is this prop not being used?

It is being used but on line 25 the parameter is not being return and esLint throws a warning for that

Extheoisah commented on src/components/sideBarContentEdit/SelectField.tsx:
```diff
@@ -293,3 +305,85 @@ export const SingleSelectField = ({
+
+  for (const obj of objArray) {
+    if (obj.slug === path[depth - 2]) {
+      if (path.length === 2 && depth === 2) {
```
Can you put a comment on why we're doing this before the recursion? Its entirely clear

0tuedon replied:
> Can you put a comment on why we're doing this before the recursion? Its entirely clear
for line 327? 
reason for doing , we want to check the current length of the array for example advancing-bitcoin/2019 would be of length 2,  and if path if found we return the nestDir or an empty array if it's not two we slice it again

Extheoisah commented on src/components/sideBarContentEdit/SidebarContentEdit.tsx:
```diff
@@ -38,7 +70,45 @@ const SidebarContentEdit = ({
+    }
+  }, [data.content.loc, directoryPaths, initialCount]);
+  useEffect(() => {
+    const convertDirStructure = (
```
Does this create a new mutated instance ofthe directiory or it duplicates it?

0tuedon replied:
> Does this create a new mutated instance ofthe directiory or it duplicates it?

it creates a mutated instance.

Extheoisah commented on src/pages/api/github/dir.ts:
```diff
@@ -0,0 +1,56 @@
+  path?: string
+) {
+  const auth = await octokit.request("GET /user");
+  // Fork the repository
```
Remove this comment. This doesn't fork the repo

0tuedon replied:
Sure

0tuedon commented:
> Left a few comments. I need to test this pr and give feedback

Noted

Emmanuel-Develops commented on src/components/sideBarContentEdit/SelectField.tsx:
```diff
@@ -293,3 +305,85 @@ export const SingleSelectField = ({
+  };
+
+  // remove previously selected option from list
+  const newAutoCompleteList =
```
This is for selecting multiple items. Selecting location is single use so there's no need to remove previous items from the list

adamjonas commented:
This isn't ready. Tried adding a new dir a couple of different ways.

https://github.com/bitcointranscripts/transcription-review-front-end/assets/755825/496c4870-1254-4d0d-91e5-c2d2fe5a8d3a

https://github.com/bitcointranscripts/transcription-review-front-end/assets/755825/630363b0-93f4-4497-a5b6-48968381b57b

0tuedon commented:
https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/b9707b19-92f0-4644-a1b2-3cf75271feb4

A new look on the select directories flow @Extheoisah  @Emmanuel-Develops  @BalogunofAfrica . Please review

adamjonas commented:
> You can add a new directory, demo is below, i think it might be the choice of words or the way of adding it?

I see. Yeah, that's not very intuitive. I liked the add new dir prompt. We'll also want to add an "are you sure?" modal to discourage people from doing this often (but not needed in a v1).

Emmanuel-Develops commented:
@adamjonas we are taking a different approach with the logic for selection
I have a figma design [here](https://www.figma.com/file/82tn8JtimD4nYmAkMo85hM/Nested-select-UI?type=design&node-id=0%3A1&mode=design&t=g96mzuneG4ASIqum-1)
Showing a simple structure where users can easily navigate directories. As well as create arbitrary ones.
@0tuedon has complete freedom over the bits and pieces of its implementation

My mistake for not sharing the document earlier so the team could critic the doc for improvements

0tuedon commented:
> > You can add a new directory, demo is below, i think it might be the choice of words or the way of adding it?
> 
> I see. Yeah, that's not very intuitive. I liked the add new dir prompt. We'll also want to add an "are you sure?" modal to discourage people from doing this often (but not needed in a v1).
> 
> Why did you stray from the previous design? I thought it functioned fine, it just had saving issues.

for the confirmation modal, i would be adding it. 
Also  we took a different approach from the previous designs. reason being it create and new instance of the dropdown for each selection.

0tuedon commented:
> > > @0tuedon not all the directories loading?
> > > e.g. chaincode-labs has no sub-directories to select.
> > > Removing the ability to add a new directory is a loss. I'm open to it, but I preferred the old way of adding a new dir.
> > 
> > 
> > You can add a new directory, demo is below, i think it might be the choice of words or the way of adding it?
> > Screen.Recording.2023-07-25.at.16.43.41.mov
> > so when a directory is selected it saves and updates the transcript before getting the nested directories
> > <img alt="Screenshot 2023-07-25 at 16 46 03" width="466" src="https://user-images.githubusercontent.com/90271995/255957613-aa546ea3-4b91-4ea6-b5eb-ef44c839de93.png">
> 
> Looks good! a couple of comments on this:
> 
> * the back button is only visible if you're not in the root directory
> * the text next to the back button can be more pronounced (try bold but grey) as that is the current path the user is in  delimited by "/" as they go further or the current dir the user is at (which is the implementation you have here) try either and see what makes the best UX
> * switch text in the last button from "select" to "add" if path doesn't exist (path from github/ tree you built in state). Adding this arbitrary path should have a confirmation popup.
> * also the back button duo-tone could be a bit distracting, try a single background or just the arrow
> * align the options ("chaincode-podcast", "chaincode-residency") vertically to the folder ("chaincode-labs"). This references the second point which helps separate current directory from subdirectories under it

Made an update to add confirmation modal and the listed minor changes
https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/9156f27d-ff69-44f5-bf69-3966d924e31c

0tuedon commented:
@Emmanuel-Develops Prior to our discussion, I have been able to implement the changes request, please review

https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/b0a6b9d8-42a0-4239-a811-a6360b0e0664

This video is for and found path


https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/f0be07eb-0e4d-4b32-b4aa-e1d82bddca03

This is for the custom path

adamjonas commented:
@0tuedon the commit history isn't right here. Can you rebase your changes on the dev branch and repush please?

0tuedon commented:
> @0tuedon the commit history isn't right here. Can you rebase your changes on the dev branch and repush please?

Alright

0tuedon commented:
> @0tuedon the commit history isn't right here. Can you rebase your changes on the dev branch and repush please?

i  think i have updated the branch and rebased this already. although i just made an update now.

0tuedon commented:
> @0tuedon the commit history isn't right here. Can you rebase your changes on the dev branch and repush please?

PR #145   fixes the commit history on this
